test/units/parsing/test_ajson.py

```diff
@@ -20,3 +25,159 @@ def test_AnsibleJSONDecoder_vault():
+                      '0316337393263616439\n646539373134633963666338613632666334'
+                      '65663730303633323534363331316164623237363831\n35363335613'
+                      '93238370a313330316263373938326162386433313336613532653538'
+                      '376662306435\n3339\n')
+
+        expected_1 = (u'$ANSIBLE_VAULT;1.1;AES256\n34646264306632313333393636316'
+                      '562356435376162633631326264383934326565333633366238\n3863'
+                      '373264326461623132613931346165636465346337310a32643431383'
+                      '0316337393263616439\n646539373134633963666338613632666334'
+                      '65663730303633323534363331316164623237363831\n35363335613'
+                      '93238370a313330316263373938326162386433313336613532653538'
+                      '376662306435\n3338\n')
+
+        # previously check that test data is a suitable type:
+        assert(isinstance(data_0, AnsibleVaultEncryptedUnicode))
+        assert(isinstance(data_1, AnsibleVaultEncryptedUnicode))
+
+        # main assertions:
+        assert(ansible_json_encoder.default(data_0) == {'__ansible_vault': expected_0})
+        assert(ansible_json_encoder.default(data_1) == {'__ansible_vault': expected_1})
+
+    @pytest.mark.parametrize('test_input,expected', [(unsafe(), ''), ])
+    def test_ansible_unsafe(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for passing AnsibleUnsafe objects
+        to AnsibleJSONEncoder.default()
+        """
+        # does not make sense to implement objects of class:
+        # class AnsibleUnsafe(object):
+        #   __UNSAFE__ = True
+        # See lib/ansible/utils/unsafe_proxy.py
+        pass
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 'first'}, {1: 'first'}),
+            ({2: 'second'}, {2: 'second'}),
+        ]
+    )
+    def test_default_encoder(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for the default encoder of json.JSONEncoder superclass
+        by passing objects of different classes that are not tested above
+        """
+        assert(ansible_json_encoder.default(test_input) == expected)
+
+    @pytest.mark.xfail(raises=TypeError)
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            (1, 1),
+            (1.1, 1.1),
+            ('string', 'string'),
+            ([1, 2], [1, 2]),
+            (True, True),
+            (None, None),
+            (set('set'), ''),
+        ]
+    )
+    def test_default_encoder_unserializable(self, ansible_json_encoder, test_input, expected):
+        """
+        Test for the default encoder of json.JSONEncoder superclass
+        by passing unserializable objects of different classes.
+        It must fail with TypeError 'object is not serializable'
+        """
+        assert(ansible_json_encoder.default(test_input) == expected)
```

Comment:
Check for the exception in the test body instead of using a decorator. Also, since you're testing for an exception, you only need to define the input and not compare against expected:

```suggestion
        with pytest.raises(TypeError):
            ansible_json_encoder.default(test_input)
```

Reply:
agreed, it is more obvious. fixed